### PR TITLE
feat(ui): Sanitize RequestError paths

### PR DESCRIPTION
**Pull request by billyvg**
This will sanitize the URLs in our RequestError exceptions that get captured to Sentry. It will replace organization, project, and team slugs with a placeholder so that it is not exposed in the exception message.

We are doing this to specifically make our Issues Stream more free of customer data.

### CODE REVIEW

File: static/app/utils/requestError/sanitizePath.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export function sanitizePath(path: string) {
+  return path.replace(
+    /(?<start>.*?)\/(?<type>organizations|projects|teams)\/(?<primarySlug>[^/]+)\/(?<contentType>[^/]+\/)?(?<tertiarySlug>[^/]+\/)?(?<end>.*)/,
+    function (...args) {
+      const matches = args[args.length - 1];
+
+      // console.log({matches});
+      // console.log(args);
```
Comment (dashed) [inline static/app/utils/requestError/sanitizePath.tsx:18]: stray console.log

Comment (billyvg) [reply static/app/utils/requestError/sanitizePath.tsx]: Oops thanks

**Review by dashed — APPROVED** [review]

File: static/app/utils/requestError/sanitizePath.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Remove organization slugs from the path - we do not want them displayed in the Issues Stream (having them in issue details is ok)
```
Comment (evanpurkhiser) [inline static/app/utils/requestError/sanitizePath.tsx:2]: More than just org slugs right?

File: static/app/utils/requestError/sanitizePath.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const SHORTENED_TYPE = {
+  organizations: 'org',
+  projects: 'project',
+  teams: 'team',
+};
+
+export function sanitizePath(path: string) {
+  return path.replace(
+    /(?<start>.*?)\/(?<type>organizations|projects|teams)\/(?<primarySlug>[^/]+)\/(?<contentType>[^/]+\/)?(?<tertiarySlug>[^/]+\/)?(?<end>.*)/,
+    function (...args) {
```
Comment (evanpurkhiser) [inline static/app/utils/requestError/sanitizePath.tsx:14]: Any reason you're writing this as a `function ()` over a fat arrow function?

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/utils/requestError/sanitizePath.tsx (hunk 3 above)
Comment (billyvg) [reply static/app/utils/requestError/sanitizePath.tsx:14]: Nope

**Pull request merged**